OutboundCallReminder/src/main/java/com/communication/outboundcallreminder/OutboundCallReminder.java

```diff
@@ -0,0 +1,362 @@
+            Response<AddParticipantResult> response = callConnection.addParticipantWithResponse(participant, alternateCallerId, operationContext, null);
+            Logger.logMessage(Logger.MessageType.INFORMATION, "addParticipantWithResponse -- > " + getResponse(response));
+        }
+
+        Boolean addParticipantCompleted = false;
+        try {
+            addParticipantCompleted = addParticipantCompleteTask.get();
+        } catch (InterruptedException ex) {
+            Logger.logMessage(Logger.MessageType.ERROR, "Failed to add participant InterruptedException -- > " + ex.getMessage());
+        } catch (ExecutionException ex) {
+            Logger.logMessage(Logger.MessageType.ERROR,"Failed to add participant ExecutionException -- > " + ex.getMessage());
+        }
+
+        return addParticipantCompleted;
+    }
+
+    private void registerToAddParticipantsResultEvent(String operationContext) {
+        addParticipantCompleteTask = new CompletableFuture<Boolean>();
+
+        NotificationCallback addParticipantReceivedEvent = ((callEvent) -> {
+            AddParticipantResultEvent addParticipantsUpdatedEvent = (AddParticipantResultEvent) callEvent;
+            OperationStatus operationStatus = addParticipantsUpdatedEvent.getStatus();
+            if (operationStatus.equals(OperationStatus.COMPLETED)) {
+                Logger.logMessage(Logger.MessageType.INFORMATION, "Add participant status -- > " + operationStatus);
+                addParticipantCompleteTask.complete(true);
+            } else if (operationStatus.equals(OperationStatus.FAILED)) {
+                addParticipantCompleteTask.complete(false);
+            }
+            EventDispatcher.getInstance().unsubscribe(CallingServerEventType.ADD_PARTICIPANT_RESULT_EVENT.toString(),
+                    operationContext);
+        });
+
+        // Subscribe to event
+        EventDispatcher.getInstance().subscribe(CallingServerEventType.ADD_PARTICIPANT_RESULT_EVENT.toString(),
+                operationContext, addParticipantReceivedEvent);
+    }
+
+    public final String userIdentityRegex = "8:acs:[0-9a-fA-F]{8}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{12}_[0-9a-fA-F]{8}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{12}";
+    public final String phoneIdentityRegex = "^\\+\\d{10,14}$";
+
+    private CommunicationIdentifierKind getIdentifierKind(String participantnumber) {
+        // checks the identity type returns as string
+        return ((Pattern.matches(userIdentityRegex, participantnumber)) ? CommunicationIdentifierKind.UserIdentity
+                : (Pattern.matches(phoneIdentityRegex, participantnumber)) ? CommunicationIdentifierKind.PhoneIdentity
+                        : CommunicationIdentifierKind.UnknownIdentity);
+    }
+
+    public String getResponse(Response<?> response)
+    {
+        String responseString = null;
```

Comment:
nit: may be use a StringBuilder class here, that will be more efficient for concatenating